frontend/src/containers/pages/ManageGroups/ManageGroupsPage.jsx

```diff
@@ -56,9 +72,38 @@ export default function ManageGroupsPage() {
     console.log("File uploaded:", event.target.files[0]);
   };
 
-  function download() {
-    console.log("downloading current group config as .CSV");
-  }
+  const convertToCSV = (data) => {
+    const headers = "email,first name,last name,group number,playable link\n";
+    let csv = headers;
+
+    data.forEach((row) => {
+      const entries = Object.entries(row);
+      entries.forEach(([, value]) => {
```

Comment:
Note that the response of the endpoint you created seems to be

```json
[
    {
        "users": [
            {
                "email": "abc@gmail.com",
                "name": "abc aba",
                "role": "doctor",
                "group": "1"
            },
            {
                "email": "xyz@gmail.com",
                "name": "xyz xyx",
                "role": "nurse",
                "group": "1"
            }
        ],
        "path": [],
        "_id": "66529b1cd689be37e01e78c3",
        "notes": {},
        "scenarioId": "65e45789a86f985e1c5af28f",
        "__v": 0
    }
]
```

Currently, the following code produces:
![image](https://github.com/UoaWDCC/VPS/assets/100455176/1469ada0-beb0-4b9e-a3f7-d6af306ba4e7)

This differs from the expected 2 rows due to how the response body is processed.

**In fact, AC 2 and 3 seem to be misinformed.**

For reference, I would have expected something along the lines of:
```csv
email, name, role, group_number, link
abc@gmail.com, abc aba, doctor, 1, https://vps-dev.wdcc.co.nz/play/65e45789a86f985e1c5af28f 
xyz@gmail.com, xyz xyx, nurse, 1, https://vps-dev.wdcc.co.nz/play/65e45789a86f985e1c5af28f 
```
(Note that the link would be the same for everyone due to how we ended up architecting the `/play/:scenarioId` page)

Ultimately, this is a failure in ticket definition - something to mention in the next retro perhaps!
For this ticket, the end point is great - and it's just minor post processing changes (that _hopefully_ should be quite easy).

Reply:
See my comment on the issue

Reply:
Fixed aforementioned issues in this comment